crates/templates/src/manager.rs

```diff
@@ -468,21 +468,104 @@ mod tests {
         PathBuf::from(crate_dir).join("tests")
     }
 
+    // A template manager in a temp dir.  This exists to ensure the lifetime
+    // of the TempDir
+    struct TempManager {
+        temp_dir: tempfile::TempDir,
+        manager: TemplateManager,
+    }
+
+    impl TempManager {
+        fn new() -> Self {
+            let temp_dir = tempdir().unwrap();
+            let store = TemplateStore::new(temp_dir.path());
+            let manager = TemplateManager { store };
+            Self { temp_dir, manager }
+        }
+
+        async fn new_with_this_repo_templates() -> Self {
+            let mgr = Self::new();
+            mgr.install_this_repo_templates().await;
+            mgr
+        }
+
+        async fn install_this_repo_templates(&self) -> InstallationResults {
+            let source = TemplateSource::File(project_root());
+            self.manager
+                .install(&source, &InstallOptions::default(), &DiscardingReporter)
+                .await
+                .unwrap()
+        }
+
+        async fn install_test_data_templates(&self) -> InstallationResults {
+            let source = TemplateSource::File(test_data_root());
+            self.manager
+                .install(&source, &InstallOptions::default(), &DiscardingReporter)
+                .await
+                .unwrap()
+        }
+
+        fn store_dir(&self) -> &Path {
+            self.temp_dir.path()
+        }
+    }
+
+    impl std::ops::Deref for TempManager {
+        type Target = TemplateManager;
+
+        fn deref(&self) -> &Self::Target {
+            &self.manager
+        }
+    }
+
+    fn run_options(
```

Comment:
I don't really find this function self explanatory. Perhaps some small docs can make this clearer should others have any confusion like I did.